delfin/drivers/dell_emc/vnx_block/alert_handler.py

```diff
@@ -0,0 +1,145 @@
+# Copyright 2020 The SODA Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import time
+
+import six
+from oslo_log import log as logging
+from oslo_utils import units
+
+from delfin import exception
+from delfin.common import constants
+from delfin.i18n import _
+
+LOG = logging.getLogger(__name__)
+
+
+class AlertHandler(object):
+    """Alert handling functions for vnx stor driver"""
+
+    OID_MESSAGECODE = '1.3.6.1.4.1.1981.1.4.5'
+    OID_DETAILS = '1.3.6.1.4.1.1981.1.4.6'
+
+    # Translation of trap severity to alert model severity
+    SEVERITY_MAP = {"76": constants.Severity.CRITICAL,
+                    "75": constants.Severity.MAJOR,
+                    "74": constants.Severity.MINOR,
+                    "73": constants.Severity.WARNING,
+                    "72": constants.Severity.WARNING,
+                    "77": constants.Severity.FATAL,
+                    "71": constants.Severity.INFORMATIONAL,
+                    "70": constants.Severity.INFORMATIONAL}
+
+    # Attributes expected in alert info to proceed with model filling
+    _mandatory_alert_attributes = (
+        OID_MESSAGECODE,
+        OID_DETAILS
+    )
+
+    def __init__(self, navi_handler=None):
+        self.navi_handler = navi_handler
+
+    @staticmethod
+    def parse_alert(alert):
+        try:
+            alert_model = dict()
+            # These information are sourced from device registration info
+            alert_model['alert_id'] = alert.get(AlertHandler.OID_MESSAGECODE)
+            alert_model['alert_name'] = alert.get(AlertHandler.OID_DETAILS)
+            alert_model['severity'] = AlertHandler.SEVERITY_MAP.get(
```

Comment:
OID  1.3.6.1.6.3.1.1.4.1(snmpTrapOID) contains the severity value in trap. It can be directly used to get severity. SEVERITY MAP can be removed
Reference:
![image](https://user-images.githubusercontent.com/30930862/106476330-f4e9c880-64cc-11eb-85cb-4ac43a3beaf4.png)

Reply:
Modification of trap alarm level value method